### DISCUSSION TITLE
Provide an API endpoint for IP List for GitHub Apps API

### DISCUSSION THREAD

**Discussion by afinetooth** [community/community#61975] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

We'd like to request a feature that would allow us to update the **IP list** associated with our **GitHub App**.

Right now, this doesn't seem to be available in the current **Apps API version**:
https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28

**Use case**:
We run a CI/CD related service through a GitHub App-based integration with GitHub. For customers that use IP restriction on their accounts, they subscribe to the IP list on our app. However, we our server fleet (and their IP addresses) change regularly and we need a way to synchronize that list with our app's IP List.

Right now, the only way is manual. It's hard to keep up and this is breaking our service for customers with IP restrictions.

If anyone has any ideas for a workaround, please let us know.

Right now, besides manual updates, our only option appears to be purchasing a block of static IPs.

Thanks.

**Comment by sandroamiridze** [comment]
Dynamic DNS: Instead of relying on static IP addresses, you can consider setting up Dynamic DNS (DDNS) for your server fleet. DDNS allows you to associate a domain name with a changing IP address. There are services and tools available that can automatically update the DNS records when the IP addresses of your servers change. This way, your GitHub App can use the domain name instead of static IP addresses, making it more flexible to adapt to IP changes.

    Webhooks for IP Updates: Implement a webhook mechanism in your GitHub App so that your servers can notify the App whenever their IP addresses change. The webhook can trigger an update to the IP list associated with your GitHub App in response to these changes.

    Scheduled IP List Updates: Develop a system that periodically checks the IP addresses of your server fleet and updates the IP list associated with your GitHub App. You can schedule this update to run at regular intervals, reducing the need for manual intervention.

    Automated API Calls: If the GitHub API allows for updating the IP list associated with your GitHub App, you can create a script or automation that fetches the current IP addresses of your servers and uses the GitHub API to update the IP list accordingly.

    Feature Request: Reach out to GitHub Support or the GitHub Developer Community to request the addition of the feature you need. Explain your use case and how such functionality would benefit your integration. GitHub actively listens to user feedback, and your request might influence future updates to the API.

Keep in mind that these are just general suggestions, and the best approach would depend on the specific details of your GitHub App and infrastructure. It's essential to thoroughly test any solution you implement to ensure it aligns with your security requirements and doesn't compromise the functionality of your service.
